### ISSUE TITLE
Proposal for some UI/UX improvements in the replay banner

### ISSUE THREAD

**Issue by machawk1** [oduwsdl/ipwb#465]
![banner](https://user-images.githubusercontent.com/2514780/43408616-ccce3a80-93ef-11e8-98f9-e74e042364a6.gif)

In #462 @ibnesayeed made some **_great_** updates to the ipwb replay banner to vastly improve its aesthetics and functionality, but I think there are a few things we can do to make it more useable. I am hoping we can use this ticket for discussing particular improvements. Below I have provided an initial set.

- [x] Highlight forward/back buttons on hover to provide an additional selected context -- redundant of the pointer icon but visually useful.
- [x] Dim forward/backward buttons when not clickable (currently just indicated by no-pointer on hover)
- [X] Prevent some series of clicks from causing the UI elements to appear "selected" (see GIF) (_see resolution below_)
- [ ] Provide some indication with the human-readable date that a hover will reveal the more specific datetime
- [x] Change the Reconstruct icon to ipwb logo
- [x] Add some event (go to ipwb replay home?) when the banner logo is clicked

**Comment by ibnesayeed** [comment]
Most of these are addressed in the upstream repo now.

**Comment by machawk1** [comment]
@ibnesayeed I am looking forward to you merging the upstream Reconstructive changes to ipwb when you get the chance.

**Comment by ibnesayeed** [comment]
> Prevent sone series of clicks from causing the UI elements to appear "selected" (see GIF)

I thought this was handled, is it still happening? If so, then how to reproduce it?

> Provide some indication with the human-readable date that a hover will reveal the more specific datetime

I think that will make the UI noisy. In presence of the natural language time version, highlighting it is not essential.

**Comment by machawk1** [comment]
>  If so, then how to reproduce it?

I can reproduce it with the 5mementos WARC by going to the earliest memento, clicking the right arrow icon 6 times, and noting that the logo highlights.

**Comment by machawk1** [comment]
@ibnesayeed I noticed that Reconstructive expects an absolute URI for the logo. Is there any way to surface the host/port info to this configuration? Ideally, a relative URI or one starting with a "/" (e.g., `/webui/images/ipwbLogo.png`) would be an easier to supply as a default.

**Comment by ibnesayeed** [comment]
> I can reproduce it with the 5mementos WARC by going to the earliest memento, clicking the right arrow icon 6 times, and noting that the logo highlights.

I can fix that, but should we?

**Comment by machawk1** [comment]
> but should we

Yes. Is there a reason we should not? I think of the buttons as interactive elements and not text, so the behavior with the superfluous click making the button seem like text is unintuitive.

**Comment by ibnesayeed** [comment]
While I know (and tested) a fix for it, I was avoiding it to not worry too much about natural browser behavior. For example, if you go to any web page where there is an image and some empty space next to it (e.g., the rendered README page of this repo with the logo and workflow diagram). Then click three times in succession next to the image in the empty area, but not on the image. You will see the image will be highlighted although the clicks were not performed on the image (the row does not belong to the image as images are by default inline elements). This is precisely what is happening here in the banner too. We can prevent it by declaring the logo image to be non-selectable, but do we really want to do that?

**Comment by machawk1** [comment]
@ibnesayeed Let's not dwell on further meeting the "Prevent some series of clicks...". I think your improvements of preventing the selection of the rest of the banner elements (current master) is sufficient.

**Comment by ibnesayeed** [comment]
Once #475 is merged, changing logo with the IPWB one will not make the banner look ugly. Actual change of the logo should be tracked in a separate ticket as it is local to the IPWB, not tied to the Reconstructive code which has already implemented its part.

**Comment by ibnesayeed** [comment]
>  Provide some indication with the human-readable date that a hover will reveal the more specific datetime

Also, I am not sure if we should do anything about it as I expressed my views on it in https://github.com/oduwsdl/ipwb/issues/465#issuecomment-409120633.

**Comment by machawk1** [comment]
I encountered a need for more specific dates when iterating through the mementos within the 5mementos sample WARC. There are a few mementos that are from a few years ago around the same time. The banner simply states "Captured about 5 years ago" for the date. That is not sufficiently descriptive.

What if we were to allow an option to display more precise dates in place of fuzzy dates? Human readability becomes far less useful when it is difficult to distinguish captures that resolve to the same "human readable date string".

**Comment by ibnesayeed** [comment]
I think more detailed and precised datetime information can be made available by adding raw request and response headers in the expanded banner. The FAB mode can still enjoy being minimal. Less precised human readable dates are still better than Date strings that are difficult for human to parse at a glance. If they do want precise time, the tooltip is there to help (that's how a very popular library moment.js does it too).

**Comment by machawk1** [comment]
> Provide some indication with the human-readable date that a hover will reveal the more specific datetime

will need further consideration and discussion. Closing this ticket.
